js/router/BRouter.js

```diff
@@ -231,6 +236,39 @@ L.BRouter = L.Class.extend({
         return lonlats;
     },
 
+    _getLonLatsNameString: function(latLngNames) {
+        var s = '';
+        for (var i = 0; i < latLngNames.length; i++) {
+            s += this._formatLatLng(latLngNames[i].latlng);
+            s += L.BRouter.NUMBER_SEPARATOR;
+            s += latLngNames[i].name;
+
+            if (i < latLngNames.length - 1) {
+                s += L.BRouter.GROUP_SEPARATOR;
+            }
+        }
+        return s;
+    },
+
+    _parseLonLatNames: function(s) {
+        var groups,
+            part,
+            lonlatnames = [];
+
+        if (!s) {
+            return lonlatnames;
+        }
+
+        groups = s.split(L.BRouter.GROUP_SEPARATOR);
+        for (var i = 0; i < groups.length; i++) {
+            // lng,lat,name
+            part = groups[i].split(L.BRouter.NUMBER_SEPARATOR);
+            lonlatnames.push({ latlng: L.latLng(part[1], part[0]), name: decodeURI(part[2]) });
```

Comment:
I think for just decoding parts of a URL this should be `decodeURIComponent` instead of `decodeURI`.

Reply:
indeed, my bad :).